keygen/src/keygen.rs

```diff
@@ -597,7 +664,13 @@ fn do_main(matches: &ArgMatches<'_>) -> Result<(), Box<dyn error::Error>> {
                                 found.load(Ordering::Relaxed),
                             );
                         }
-                        let keypair = Keypair::new();
+                        let (keypair, phrase) = if use_mnemonic {
+                            let mnemonic = Mnemonic::new(mnemonic_type, language);
+                            let seed = Seed::new(&mnemonic, &passphrase);
+                            (keypair_from_seed(seed.as_bytes()).unwrap(), mnemonic.phrase().to_string())
```

Comment:
@t-nelson , can you review this as well?
It certainly works.

Reply:
> @t-nelson , can you review this as well? It sounded like you might be envisioning something a little different
> This certainly works.

It turns out my vision doesn't work.  I was thinking BIP39 ran the KDF on the entropy bytestring, but it runs it directly on the mnemonic string.  The former could be much faster as we wouldn't have to build all of the invalid mnemonic strings, alas...